.github/workflows/scripts/test_init_scripts.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+EESSI_VERSION="2023.06"
+export LMOD_PAGER=cat
+
+# initialize assert framework
+if [ ! -d assert.sh ]; then
+	echo "assert.sh not cloned."
+	echo ""
+	echo "run \`git clone https://github.com/lehmannro/assert.sh.git\`"
+	exit 1
+fi
+. assert.sh/assert.sh
+
+SHELLS=$@
+
+for shell in ${SHELLS[@]}; do
+	echo = | awk 'NF += (OFS = $_) + 100'
+	echo  RUNNING TESTS FOR SHELL: $shell
+	echo = | awk 'NF += (OFS = $_) + 100'
+
+	# TEST 1: Source Script and check Module Output
+	assert "$shell -c 'source init/lmod/$shell' 2>&1 " "EESSI/$EESSI_VERSION loaded successfully"
+	# TEST 2: Check if module overviews first section is the loaded EESSI module
+	MODULE_SECTIONS=($($shell -c "source init/lmod/$shell 2>/dev/null; module ov 2>&1 | grep -e '---'"))
+	assert "echo ${MODULE_SECTIONS[1]}" "/cvmfs/software.eessi.io/versions/$EESSI_VERSION/software/linux/x86_64/intel/haswell/modules/all"
```

Comment:
You've got no guarantees what architecture you will end up with so better to override the architecture to give you something predictable.

Reply:
```
export EESSI_ARCHDETECT_OPTIONS="x86_64/generic"
```